feat(k8s): routed ips (#3154)
scaleway/docs-content: containers/kubernetes/reference-content/move-kubernetes-nodes-routed-ip.mdx

---
meta:
  title: Moving Kubernetes nodes to routed IPs
  description: Safely moving Kubernetes nodes to routed IPs
content:
  h1: Moving Kubernetes nodes to routed IPs

>   paragraph: Safely Migrating Kapsule clusters to routed IPs

Reviewer: Don't description and paragraph usually have the same text?

Author: ```suggestion
  paragraph: Safely moving Kubernetes nodes to routed IPs
```

tags: routed-ip ip-mobility kubernetes kapsule kosmos
dates:
  validation: 2023-04-22
  posted: 2023-04-22
categories:
  - kubernetes
---

As part of our ongoing efforts to enhance infrastructure capabilities and provide better services to our customers, Scaleway is [introducing routed IPs for all products](https://www.scaleway.com/en/news/routed-ips-are-coming-to-all-scaleway-products/), including Kubernetes (K8s) worker nodes.

> One of the standout benefits of routed IPs is their support for [IP mobility](https://www.scaleway.com/en/docs/compute/instances/concepts/#ip-mobility), streamlining IP management and movement within the Scaleway ecosystem. By simplifying the process of reallocating IPs across different products, routed IPs empower users with unprecedented flexibility and control over their network infrastructure.

Reviewer: ```suggestion

One of the standout benefits of routed IPs is their support for [IP mobility](https://www.scaleway.com/en/docs/compute/instances/concepts/#ip-mobility), streamlining IP management and movement within the Scaleway ecosystem. By simplifying the process of reallocating IPs across different products, routed IPs empower users with unprecedented flexibility and control over their network infrastructure.
```

Additionally, routed IPs unlock the ability to attach multiple IPv4 addresses to a single Instance, enabling specific functionalities such as IP failover. While this feature may have limited applicability within Kubernetes environments, it offers invaluable flexibility and resilience across other Scaleway products.

<Message type="note">
  Routed IPs provide primary support for IPv6, ensuring direct connectivity for IPv6 addresses with Instances.

>   However, it is important to note that while routed IPs offer this capability, IPv6 support for Kapsule is not yet available due to compatibility limitations with existing Container Network Interfaces (CNIs).

Reviewer: ```suggestion
  However, IPv6 support for Kapsule is not yet available due to compatibility limitations with existing Container Network Interfaces (CNIs).
```

</Message>

<Message type="important">
    NAT IPs will reach end-of-life in Q4/2024. After this date, Scaleway will organize migrations for all clusters to transition to routed IPs.
    Migrations will be planned region by region, with specific dates to be announced.
</Message>

## Moving your Kubernetes nodes to routed IPs

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- Set up a [Kubernetes Kapsule cluster](/containers/kubernetes/how-to/create-cluster)

<Message type="note">

>   Newly created Kapsule clusters are deployed with IP mobility enabled by default. For existing clusters you can launch the migration to routed IPs manually.

Reviewer: ```suggestion
  Newly created Kapsule clusters are deployed with IP mobility enabled by default. For existing clusters, you can launch the migration to routed IPs manually.
```

> </MEssage>

Reviewer: ```suggestion
</Message>
```

### Using the Scaleway console

1. Click **Kubernetes** in the **Containers** section of the [Scaleway console](https://console.scaleway.com) side menu. The Kubernetes Kapsule overview displays.
2. Click the name of the cluster you want to access. The cluster overview page displays.

> 3. Click the **Network tab**. The cluster network information displays.

Reviewer: ```suggestion
3. Click the **Network** tab. The cluster network information displays.
```

4. Click **Move to routed IP** in the **IP address type** section of the page. A pop-up displays.
5. Confirm the move of your worker nodes to routed IPs by typing **MOVE**, then click **Move to routed IPs** to launch the process.
    <Message type="important">

>       * This operation is similar to a clster upgrade. Each node in your cluster will reboot. Ensure having more than one node ready in order to avoid downtime.

Reviewer: ```suggestion
      * This operation is similar to a cluster upgrade. Each node in your cluster will reboot. Ensure having more than one node ready in order to avoid downtime.
```

    </Message>

    The worker nodes of your Kubernetes cluster will reboot during the process and a new flexible IP will be assigned to each worker node.
    Once the migration has completed, the IP address type of your cluster will change to **Routed flexible IPs**.

### Using the API

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- Created a [Kubernetes cluster](/containers/kubernetes/how-to/create-cluster/) that uses NAT IPs
- A valid [API key](/identity-and-access-management/iam/how-to/create-api-keys/)

1. Configure the following variables for your Kapsule cluster:
    ```
    REGION=fr-par # The region of your cluster. Can be either "fr-par", "nl-ams", or "pl-waw"
    CLUSTER_ID=b69106db-e212-4fff-9584-5da4a47217ed # The ID of your Kapsule cluster
    SCW_SECRET_KEY=196dadd9-5685-42c8-aa71-a34d46861303 # Your Scaleway secret key
    ```

> 2. Run the following command to launch the migration of exisiting Kapsule clusters to routed IPs:

Reviewer: ```suggestion
2. Run the following command to launch the migration of existing Kapsule clusters to routed IPs:
```

---
meta:
  title: Moving Kubernetes nodes to routed IPs
  description: Safely moving Kubernetes nodes to routed IPs
content:
  h1: Moving Kubernetes nodes to routed IPs
  paragraph: Safely moving Kubernetes nodes to routed IPs
tags: routed-ip ip-mobility kubernetes kapsule kosmos
dates:
  validation: 2023-04-22
  posted: 2023-04-22
categories:
  - kubernetes
---

As part of our ongoing efforts to enhance infrastructure capabilities and provide better services to our customers, Scaleway is [introducing routed IPs for all products](https://www.scaleway.com/en/news/routed-ips-are-coming-to-all-scaleway-products/), including Kubernetes (K8s) worker nodes.

> One of the standout benefits of routed IPs is their support for [IP mobility](https://www.scaleway.com/en/docs/compute/instances/concepts/#ip-mobility), streamlining IP management and movement within the Scaleway ecosystem. By simplifying the process of reallocating IPs across different products, routed IPs empower users with unprecedented flexibility and control over their network infrastructure.

Reviewer 2: ```suggestion
One of the standout benefits of routed IPs is their support for [IP mobility](/compute/instances/concepts/#ip-mobility), streamlining IP management and movement within the Scaleway ecosystem. By simplifying the process of reallocating IPs across different products, routed IPs empower users with unprecedented flexibility and control over their network infrastructure.
```

Additionally, routed IPs unlock the ability to attach multiple IPv4 addresses to a single Instance, enabling specific functionalities such as IP failover. While this feature may have limited applicability within Kubernetes environments, it offers invaluable flexibility and resilience across other Scaleway products.

<Message type="note">
  Routed IPs provide primary support for IPv6, ensuring direct connectivity for IPv6 addresses with Instances.
  However, IPv6 support for Kapsule is not yet available due to compatibility limitations with existing Container Network Interfaces (CNIs).
</Message>

<Message type="important">

>     NAT IPs will reach end-of-life in Q4/2024. After this date, Scaleway will organize migrations for all clusters to transition to routed IPs.

Reviewer 2: ```suggestion
    NAT IPs will reach end-of-life in Q4 2024. After this date, Scaleway will organize migrations for all clusters to transition to routed IPs.
```

    Migrations will be planned region by region, with specific dates to be announced.
</Message>

## Moving your Kubernetes nodes to routed IPs

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- Set up a [Kubernetes Kapsule cluster](/containers/kubernetes/how-to/create-cluster)

<Message type="note">

>   Newly created Kapsule clusters are deployed with IP mobility enabled by default. For existing clusters, you can manually launch the migration to routed IPs.

Reviewer 3: Kapsule and Kosmos clusters created after April 12th 2024 are deployed with IP mobility enabled by default ([changelog](https://www.scaleway.com/en/docs/changelog/?product=kubernetes#routed-ips-are-now-default-on-kubernetes-clusters))

Author: ```suggestion
  Kapsule and Kosmos clusters created after April 12th 2024 are deployed with IP mobility enabled by default. For existing clusters, you can manually launch the migration to routed IPs.
```

>   Newly created Kapsule clusters are deployed with IP mobility enabled by default. For existing clusters, you can launch the migration to routed IPs manually.

Reviewer 2: ```suggestion
  Newly created Kapsule clusters are deployed with IP mobility enabled by default. For existing clusters, you can manually launch the migration to routed IPs.
```

</Message>

### Using the Scaleway console

> 1. Click **Kubernetes** in the **Containers** section of the [Scaleway console](https://console.scaleway.com) side menu. The Kubernetes Kapsule overview displays.

Reviewer 3: Skip Kapsule, it can be either Kapsule or Kosmos clusters in this list

Author: ```suggestion
1. Click **Kubernetes** in the **Containers** section of the [Scaleway console](https://console.scaleway.com) side menu. The Kubernetes overview displays.
```

2. Click the name of the cluster you want to access. The cluster overview page displays.
3. Click the **Network** tab. The cluster network information displays.
4. Click **Move to routed IP** in the **IP address type** section of the page. A pop-up displays.
5. Confirm the move of your worker nodes to routed IPs by typing **MOVE**, then click **Move to routed IPs** to launch the process.
    <Message type="important">

>       * This operation is similar to a cluster upgrade. Each node in your cluster will reboot. Ensure having more than one node ready in order to avoid downtime.

Reviewer 2: ```suggestion
      * This operation is similar to a cluster upgrade. Each node in your cluster will reboot. Ensure that you have more than one node ready in order to avoid downtime.
```

      * Kubernetes clusters use dynamic IPs, meaning nodes may receive new public IPs upon reboot. Users relying on node IPs for filtering/whitelisting should take necessary precautions.
    </Message>

    The worker nodes of your Kubernetes cluster will reboot during the process and a new flexible IP will be assigned to each worker node.
    Once the migration has completed, the IP address type of your cluster will change to **Routed flexible IPs**.

### Using the API

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- Created a [Kubernetes cluster](/containers/kubernetes/how-to/create-cluster/) that uses NAT IPs
- A valid [API key](/identity-and-access-management/iam/how-to/create-api-keys/)

> 1. Configure the following variables for your Kapsule cluster:

Reviewer 3: Let's say Kubernetes cluster throughout, it can be either Kapsule or Kosmos
